### DISCUSSION TITLE
seems create trigger with "when condition" doesn't work in dashboard ui

### DISCUSSION THREAD

**Discussion by devhandler** [supabase/supabase#15627] [Questions] [answered]
it always complain syntax error when you specify a "when" condition when creating a trigger. anyone has similar experience?

**Comment by GaryAustin1** [comment]
Can you give more info...
```
CREATE TRIGGER check_update
    BEFORE UPDATE ON comments
    FOR EACH ROW
    WHEN (OLD.body != null)
    EXECUTE FUNCTION defaulttest();
```
 Works in the SQL editor.

**Reply by devhandler** [reply]
the below shows invalid syntax

```
create or replace trigger before_insert_intro_photos
  before insert on storage.objects
  for each row 
  when new.bucket_id = 'profile'
  execute function public.check_photo_num_limit('profile')
```

hmm....maybe it's another issue. when i tried to run it via pgadmin, pgadmin complains :
```
ERROR:  syntax error at or near "new"
```

maybe because we cannot reference bucket_id column?

**Reply by GaryAustin1** [reply, marked as answer]
Probably when () with parenthesis.

**Reply by devhandler** [reply]
you are right! this is so helpful! thank you.

**Reply by devhandler** [reply]
@GaryAustin1 do you mind taking a look at the trigger function called? the trigger func checks whether a user has already had 3 uploads, if so, return null and abort the insert. but now even if a user has only 1 upload, somehow i still can't insert. i even tried manually insert via pgadmin by running the below. it just returns running successfully without any error, but no row is inserted. i tried add raise notice in the trigger function but no notice msg sent to the client even with SET client_min_messages TO DEBUG5. so idea what's wrong here or how to debug...

```
insert into storage.objects (bucket_id, name, owner)
values('profile', 'test.jpg', '5e6104de-29fb-4646-9922-88833385b28c')
```

```
create or replace function check_photo_num_limit()
returns trigger
language plpgsql
security invoker
as $$
declare
  usedUploads int;
begin
  select count(name) into uploads
  from storage.objects
  where bucket_id = TG_ARGV[0] and owner = auth.uid();
  raise notice 'tg_argv[0] %, authuid %, uploads %', TG_ARGV[0], auth.uid(), uploads
  if uploads > 3 then 
    return null; 
  end if;
  return new;
end;
$$;
```

**Reply by GaryAustin1** [reply]
You should consider using Supabase discord, it is more friendly to these types of questions and back and forth...
I use raise log 'value = %',var; and the message goes to the postgres logs.
